presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -125,18 +123,22 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
                 partitionKeys.put(id, null);
             }
             else {
-                String partitionValue;
-                if (type.typeId() == FIXED || type.typeId() == BINARY) {
-                    // this is safe because Iceberg PartitionData directly wraps the byte array
-                    partitionValue = new String(((ByteBuffer) value).array(), UTF_8);
-                }
-                else {
-                    partitionValue = value.toString();
-                }
-                partitionKeys.put(id, partitionValue);
+                partitionKeys.put(id, partitionValueFromType(type, value));
             }
         });
 
         return Collections.unmodifiableMap(partitionKeys);
     }
+
+    private static String partitionValueFromType(Type type, Object value)
+    {
+        switch (type.typeId()) {
```

Comment:
I think you accidentally removed the case for `TIME`

Reply:
> Since in String.valueOf((Long) value), the cast to Long is redundant, the former TIME case is correctly handled by the default case, as pointed out by Intellij.

No; as mentioned in this PR comment above, the default case suffices